hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/SCMSecurityProtocolServer.java

```diff
@@ -156,6 +159,44 @@ public String getOMCertificate(OzoneManagerDetailsProto omDetails,
     }
   }
 
+
+  /**
+   * Get SCM signed certificate for SCM peer Node.
+   *
+   * @param scmNodeDetails   - SCM Node Details.
+   * @param certSignReq - Certificate signing request.
+   * @return String         - SCM signed pem encoded certificate.
+   */
+  @Override
+  public String generateSCMPeerCertificate(ScmNodeDetailsProto scmNodeDetails,
+      String certSignReq) throws IOException {
+    Objects.requireNonNull(scmNodeDetails);
+    LOGGER.info("Processing CSR for scm {}, nodeId: {}",
+        scmNodeDetails.getHostName(), scmNodeDetails.getScmNodeId());
+
+    // Check clusterID
+    if (storageContainerManager.getClusterId().equals(
+        scmNodeDetails.getClusterId())) {
+      throw new IOException("SCM ClusterId mismatch. Peer SCM ClusterId, " +
+          scmNodeDetails.getClusterId() + "leader SCM ClusterId "
+          + storageContainerManager.getClusterId());
+    }
+
+
+    Future<X509CertificateHolder> future =
+        certificateServer.requestCertificate(certSignReq,
+            KERBEROS_TRUSTED, true);
+
+    try {
+      return CertificateCodec.getPEMEncodedString(future.get());
+    } catch (InterruptedException e) {
+      Thread.currentThread().interrupt();
+      throw new IOException("generateSCMPeerCertificate operation failed. ", e);
+    } catch (ExecutionException e) {
+      throw new IOException("generateSCMPeerCertificate operation failed. ", e);
+    }
```

Comment:
Can you please extract this repeated block to a method?  Now appears three times with slight differences.

Reply:
Done